Fix lines in the software renderer

unknownbrackets opened the pull request:
This implements lines in transform mode (such that Persona 2, etc. look good), and also implements mipmapping for them (in case used.)

Also implements a very basic version of antialias, mostly so the incorrect texture UVs don't show like in #6483.

-[Unknown]

unknownbrackets committed:
SoftGPU: Implement lines in transform mode.

unknownbrackets committed:
SoftGPU: Support flat shading for lines.

unknownbrackets committed:
SoftGPU: Bias bottom right side of line drawing.

Fixes lines in Persona 2.

unknownbrackets committed:
SoftGPU: Implement simple mipmapping for lines.

Mostly so the min/mag filter is correct.

unknownbrackets committed:
SoftGPU: Implement pretend antialias.

This doesn't really smooth, but at the very least faking it makes Persona
3 have less ugly lines.

hrydgard commented on GPU/Software/Clipper.cpp:
```diff
@@ -238,8 +238,35 @@ void ProcessLine(VertexData& v0, VertexData& v1)
 		return;
 	}
+	
+	// TODO: Change logic when it's a backface (why? In what way?)
```
Surely lines don't have backfaces?

unknownbrackets replied:
Oh sorry, copy/pasted from triangles.  Will remove.

-[Unknown]

unknownbrackets committed:
SoftGPU: Clip lines entirely outside early.

Some were getting through, which may mean the line clipping isn't working
so well...
